src/components/calculator.js

```diff
@@ -1,69 +1,61 @@
-import React from 'react';
+/* eslint-disable react/destructuring-assignment */
 
+import React from 'react';
+import calculate from '../logic/calculate';
 import './calculator.css';
 
-const numbers = () => {
-  const array = [];
-  for (let i = 9; i >= 7; i -= 1) {
-    array.push(
-      <button type="button" id={i}>{i}</button>,
-    );
-  }
-  return array;
-};
-
-const number2 = () => {
-  const array = [];
-  for (let i = 6; i >= 4; i -= 1) {
-    array.push(
-      <button type="button" id={i}>{i}</button>,
-    );
+// eslint-disable-next-line react/prefer-stateless-function
+class Calculator extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { total: null, next: null, operation: null };
+    this.handleEvent = this.handleEvent.bind(this);
   }
-  return array;
-};
 
-const number3 = () => {
-  const array = [];
-  for (let i = 1; i <= 3; i += 1) {
-    array.push(
-      <button type="button" id={i}>{i}</button>,
-    );
+  handleEvent(e) {
+    // eslint-disable-next-line react/no-access-state-in-setstate
+    this.setState(calculate(this.state, e.target.textContent));
   }
-  return array;
-};
 
-// };
-// eslint-disable-next-line react/prefer-stateless-function
-class Calculator extends React.Component {
   render() {
     return (
       <div className="calculator">
         <header className="head">
-          <span>0</span>
+          <span>
+            {this.state.total}
+            {this.state.next}
+            {this.state.operation}
+          </span>
```

Comment:
[OPTIONAL] 

Great job! Your calculator is giving us the correct answers 👍🏽 , but there is an issue with the way the digits are displayed, when you enter the first number, press an operator, as you are entering the second number, the digits are concatenating with the first number, please kindly fix the issue 😄. Because normally when you enter the first number, press an operator, by the time you are entering the second number, the calculator should either display the second number after the operator or it should clear the first number but in your case it is joining the first and second number before the operator, please fix that. 123 + 456 is displaying as 123456+

Reply:
Hello Vickymarz. I will work on making the display more effective. Thank you for the highlights and review.